src/main/kotlin/io/snyk/plugin/ui/jcef/GenerateAIFixHandler.kt

```diff
@@ -0,0 +1,78 @@
+package io.snyk.plugin.ui.jcef
+
+import com.google.gson.Gson
+import com.intellij.openapi.project.Project
+import com.intellij.ui.jcef.JBCefBrowserBase
+import com.intellij.ui.jcef.JBCefJSQuery
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.cef.browser.CefBrowser
+import org.cef.browser.CefFrame
+import org.cef.handler.CefLoadHandlerAdapter
+import snyk.common.lsp.LanguageServerWrapper
+
+class GenerateAIFixHandler(private val project: Project) {
+
+    fun generateAIFixCommand(jbCefBrowser: JBCefBrowserBase): CefLoadHandlerAdapter {
+        val aiFixQuery = JBCefJSQuery.create(jbCefBrowser)
+
+        aiFixQuery.addHandler { value ->
+            val params = value.split(":")
+            val folderURI = params[0]
+            val fileURI = params[1]
+            val issueID = params[2]
+
+            // Avoids blocking the UI thread
+            CoroutineScope(Dispatchers.IO).launch {
```

Comment:
Same here - I'd prefer `runAsync`, to keep it similar to other places in the code. But should work.